Ginger/GingerCore/ALM/ZephyrEntCore.cs

```diff
@@ -0,0 +1,338 @@
+        }
+
+        public dynamic CreateNewTestPlanningFolder(long cycleId, long parenttreeid, string folderName, string folderDesc)
+        {
+            return zephyrEntRepository.CreateNewTestPlanningFolder(cycleId, parenttreeid, folderName, folderDesc);
+        }
+        public dynamic UpdateTestPlanningFolder(long cycleId, long parenttreeid, string folderName, string folderDesc)
+        {
+            return zephyrEntRepository.UpdateTestPlanningFolder(cycleId, parenttreeid, folderName, folderDesc);
+        }
+        
+        public override bool ConnectALMProject()
+        {
+            return this.ConnectALMServer();
+        }
+
+        public List<string[]> GetTCsDataSummary(int tsId)
+        {
+            return zephyrEntImportManager.GetTCsDataSummary(tsId);
+        } 
+        public override bool ConnectALMServer()
+        {
+            try
+            {
+                if (zephyrEntRepository == null)
+                {
+                    zephyrEntRepository = new Zepyhr_Ent_Repository.ZephyrEntRepository(new LoginDTO()
+                    {
+                        User = ALMCore.DefaultAlmConfig.ALMUserName,
+                        Password = ALMCore.DefaultAlmConfig.ALMPassword,
+                        AuthToken = ALMCore.DefaultAlmConfig.ALMToken,
+                        Server = ALMCore.DefaultAlmConfig.ALMServerURL
+                    });
+                }
+                Reporter.ToLog(eLogLevel.DEBUG, "Connecting to Zephyr server");
+                return Task.Run(() =>
+                {
+                    return zephyrEntRepository.IsLoginValid(
+                        new LoginDTO()
+                        {
+                            User = ALMCore.DefaultAlmConfig.ALMUserName,
+                            Password = ALMCore.DefaultAlmConfig.ALMPassword,
+                            AuthToken = ALMCore.DefaultAlmConfig.ALMToken,
+                            Server = ALMCore.DefaultAlmConfig.ALMServerURL
+                        });
+                }).Result;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Connecting to Zephyr server", ex);
+                return false;
+            }
+        }
+
+        public TestCaseResource UpdateTestCase(long treeNodeId, ActivitiesGroup activtiesGroup, List<BaseResponseItem> matchingTC)
+        {
+            return zephyrEntExportManager.UpdateTestCase(treeNodeId, activtiesGroup, matchingTC);
+        }
+
+        public TestCaseResource CreateTestCase(long treeNodeId, ActivitiesGroup activtiesGroup)
+        {
+            return zephyrEntExportManager.CreateTestCase(treeNodeId, activtiesGroup);
+        }
+
+        public override Dictionary<Guid, string> CreateNewALMDefects(Dictionary<Guid, Dictionary<string, string>> defectsForOpening, List<ExternalItemFieldBase> defectsFields, bool useREST = false)
+        {
+            throw new NotImplementedException();
```

Comment:
Does Zephyr support defects creation ? if yes we need to implement this feature like we have for other ALMs

Reply:
no, created with jira